.github/workflows/ci.yml

```diff
@@ -0,0 +1,50 @@
+name: CI Workflow
+
+on:
+  push:
+    branches:
+      - ci/setup-github-workflow
+  pull_request:
+    branches:
+      - ci/setup-github-workflow
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres:16
+        env:
+          POSTGRES_HOST_AUTH_METHOD: trust
+        ports:
+          - 5433:5432
+        options: >-
+          --health-cmd "pg_isready -U postgres" 
+          --health-interval 10s 
+          --health-timeout 5s 
+          --health-retries 5
+          
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Install Docker Compose
+        run: |
+          sudo curl -L "https://github.com/docker/compose/releases/download/v2.20.2/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
+          sudo chmod +x /usr/local/bin/docker-compose
+
+      - name: Set up Docker for Data Handler
+        run: |
+          docker-compose -f devops/dev/docker-compose.data-handler.yaml up -d --build
```

Comment:
just tested it, it doesn't work.
so if you build images and run container why you run test command inside container
second thing, why you can just run `poerty install` and then `test commands`